packages/embark/src/cmd/cmd_controller.js

```diff
@@ -216,11 +216,7 @@ class EmbarkController {
           console.dir("-- before timeout - file changed")
 
           if (fileType === 'contract' || fileType === 'config') {
-            let contractsFiles = await engine.events.request2("config:contractsFiles");
-            let compiledContracts = await engine.events.request2("compiler:contracts:compile", contractsFiles);
-            let _contractsConfig = await engine.events.request2("config:contractsConfig");
-            let contractsConfig = cloneDeep(_contractsConfig);
-            let [contractsList, contractDependencies] = await engine.events.request2("contracts:build", contractsConfig, compiledContracts);
+            await compileAndDeploySmartContracts(engine);
```

Comment:
@jrainville this logic is extracted into this function now and will be used by `run2()` and `console()` (soon `build()` once this lands and it's rebased)